sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveResolutionSuite.scala

```diff
@@ -81,9 +81,11 @@ class HiveResolutionSuite extends HiveComparisonTest {
       .toDF().registerTempTable("caseSensitivityTest")
 
     val query = sql("SELECT a, b, A, B, n.a, n.b, n.A, n.B FROM caseSensitivityTest")
-    assert(query.schema.fields.map(_.name) === Seq("a", "b", "A", "B", "a", "b", "A", "B"),
+    assert(query.schema.fields.map(_.name) === Seq("a", "B", "a", "B", "a", "b", "A", "B"),
```

Comment:
Does that matter for a case-insensitive system? 
But we do need keep the attribute name identical in the references chain. This is a workaround approach for the bug fixing, in long term, we probably need to refactor the AttributeReference `equality` for name (or take the Resolver in?).